packages/jest-config/src/normalize.js

```diff
@@ -435,6 +436,12 @@ function normalize(config, argv) {
     return newConfig;
   }, newConfig);
 
+  // If argv.json is set the coverageReporters array must be overridden
+  if (argv.json) {
+
+    newConfig.coverageReporters = ["json"];
```

Comment:
The default is `['json', 'text', 'lcov', 'clover']`. I think instead of overwriting it, we should just remove the text one:

```js
newConfig.coverageReporters = newConfig.coverageReporters.filter(reporter => reporter !== 'text');
```